### PR TITLE
feat: improve types

### PR DESCRIPTION
**Pull request by LuisMartinez15**
EX-7789

### CODE REVIEW

File: packages/x-priority-queue/src/x-priority-queue.types.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { Dictionary } from '@empathyco/x-utils';
+
+/**
+ * Adsadsadas.
```
Comment (tiborux) [inline packages/x-priority-queue/src/x-priority-queue.types.ts:4]: Fix this comment

File: packages/x-priority-queue/src/x-priority-queue.ts [hunk 2]
```diff
@@ -109,24 +87,26 @@ export class XPriorityQueue<Key = string> {
    * an array of the parametrized `Key` type.
    *
    * @returns The list of keys.
-   *
-   * @public
    */
-  public get keys(): Key[] {
+  public get keys(): SomeKey[] {
```
Comment (tiborux) [inline packages/x-priority-queue/src/x-priority-queue.ts:91]: In the constructor, change the type to be `(a: number, b: number) => boolean`

File: packages/x-priority-queue/src/x-priority-queue.ts [hunk 3]
```diff
@@ -109,24 +87,26 @@ export class XPriorityQueue<Key = string> {
...
     return this.nodes.map(({ key }) => key);
   }
 
-  /**
-   * Inserts the specified key and priority pair, with an optional metadata, into the queue.
+  /**.
+   * See {@link XPriorityQueue.(push:1)}
```
Comment (tiborux) [inline packages/x-priority-queue/src/x-priority-queue.ts:96]: Can we use just {@link XPriorityQueue.push} I dont know if it works.

File: packages/x-priority-queue/src/x-priority-queue.ts [hunk 4]
```diff
@@ -1,51 +1,25 @@
...
-   * @public
-   */
-  public readonly key: Key;
-
-  /**
-   * The number used to sort the elements in the queue.
-   *
-   * @public
-   */
+export class BaseXPriorityQueueNode<
```
Comment (tiborux) [inline packages/x-priority-queue/src/x-priority-queue.ts:10]: `priority number and data record.`

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

**Review by CachedaCodes — APPROVED** [review]

**Review by tiborux — APPROVED** [review]

**Pull request merged**